src/application_settings/container_section_base.py

```diff
@@ -0,0 +1,27 @@
+"""Base class for sections to be added to containers and container sections for configuration and settings."""
+import sys
+from dataclasses import replace
+from typing import Any, TypeVar
+
+from pydantic.dataclasses import dataclass
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    Self = TypeVar("Self", bound="ContainerSectionBase")
+
+
+@dataclass(frozen=True)
+class ContainerSectionBase:
+    """Base class for all ContainerSection classes"""
+
+    if sys.version_info >= (3, 11):
```

Comment:
So is the issue the usages of `Self` and mypy probably doesn't like that its different depending on the python version? In my opinion we should then not use `Self`, or only support Python3.11. The last solution could be to just use the `Self = TypeVar(...` such that when we stop supporting 3.10 we can just replace the line with an import.

Reply:
I really like to use Self, that is informative I think. I now have found that for 3.10 there is no problem in stating self: Self but there is a problem with stating cls: type[Self] which can be suppressed easily.
And I now understand that one cannot use a TypeAlias for a TypeVar. So it is impossible to use a TypeAlias to express 3.9 or 3.10 type notation for an optional type. I will continue to search tomorrow to see if we can get away with 3.9 notation.
I will work towards not having duplicate code.

Reply:
Did you already come across this:

https://pypi.org/project/typing-extensions/

Seems pretty promising?

Reply:
I did know about typing-extensions but not that it also has Self in it. Hooray! That solves all problems. Type checking for 3.9 is back, clean code.

Reply:
Done